live-hls.js

```diff
@@ -274,14 +291,38 @@ extractHeader = function(header, event) {
   return lines;
 };
 
-extractResourceWindow = function(mfest,duration,event) {
+calculateDatetime = function(event, resource, i) {
+  var currentDateTime;
+  var DateObject;
+  currentDateTime = resource[i].datetime.split('TIME:')[1];
+  DateObject = new Date(currentDateTime);
+  //Determine if the stream has looped yet
+  if (event.dropped > (resource.length-event.window)) {
+    //Adjust datetime according to the number of loops have occurred.
+    if (event.calculatedDuration > 0) {
+      //If we have values based on EXTINF, use the sum of all EXTINF to calculate loop
+      loopDuration = event.calculatedDuration;
+      DateObject.setSeconds(DateObject.getSeconds() + event.calculatedDuration);
+    } else {
+      //If we don't have values based on EXTINF, use targetDuration (which event.rate is based on) to calculate loop
+      DateObject.setSeconds(DateObject.getSeconds() + (Math.floor((event.dropped + event.window)/resource.length)*event.window)*event.rate);
```

Comment:
For when ES6 is a thing: Destructuring would let you extract all of these variables from the `event`:

```
const event = {
  dropped,
  window,
  rate
}

Math.floor((dropped + window) / resource.length) * window) * rate
```